### PR TITLE
Less cloning and additional pattern simplifications

### PR DESCRIPTION
**Pull request by quentinlesceller**
Similar to https://github.com/mimblewimble/grin/pull/3223.
Shouldn't change any logic.

### CODE REVIEW

File: api/src/owner_rpc.rs [hunk 1]
```diff
@@ -1344,7 +1344,7 @@ where
 
 	fn get_stored_tx(&self, tx: &TxLogEntry) -> Result<Option<TransactionV3>, ErrorKind> {
 		Owner::get_stored_tx(self, None, tx)
-			.map(|x| x.map(|y| TransactionV3::from(y)))
```
Comment (yeastplume) [inline api/src/owner_rpc.rs]: Lol

File: api/src/owner.rs [hunk 2]
```diff
@@ -2137,10 +2137,10 @@ where
...
 	///
 	/// if let Ok(p) = result {
-	///		let valid = api_owner.verify_payment_proof(None, &p);
-	///		if let Ok(_) = valid {
-	///		  //...
-	///		}
+	///     let valid = api_owner.verify_payment_proof(None, &p);
+	///     if let Ok(_) = valid {
+	///       //...
+	///     }
```
Comment (yeastplume) [inline api/src/owner.rs]: What are you running on it to tidy up doc comment formatting? Rustfmt doesn't get it

File: api/src/owner_rpc_s.rs [hunk 3]
```diff
@@ -2173,21 +2173,19 @@ where
 		let secp = secp_inst.lock();
 		let sec_key = SecretKey::new(&secp, &mut thread_rng());
 
-		let mut shared_pubkey = ecdh_pubkey.ecdh_pubkey.clone();
+		let mut shared_pubkey = ecdh_pubkey.ecdh_pubkey;
 		shared_pubkey
 			.mul_assign(&secp, &sec_key)
-			.map_err(|e| ErrorKind::Secp(e))?;
+			.map_err(ErrorKind::Secp)?;
```
Comment (yeastplume) [inline api/src/owner_rpc_s.rs]: Does this not lose the actual contents of the error?

File: api/src/owner_rpc_s.rs [hunk 4]
```diff
@@ -2173,21 +2173,19 @@ where
...
-			SecretKey::from_slice(&secp, &x_coord[1..]).map_err(|e| ErrorKind::Secp(e))?;
+		let shared_key = SecretKey::from_slice(&secp, &x_coord[1..]).map_err(ErrorKind::Secp)?;
 		{
 			let mut s = self.shared_key.lock();
 			*s = Some(shared_key);
 		}
 
-		let pub_key =
-			PublicKey::from_secret_key(&secp, &sec_key).map_err(|e| ErrorKind::Secp(e))?;
+		let pub_key = PublicKey::from_secret_key(&secp, &sec_key).map_err(ErrorKind::Secp)?;
```
Comment (yeastplume) [inline api/src/owner_rpc_s.rs]: Same here

File: impls/src/adapters/http.rs [hunk 5]
```diff
@@ -107,7 +106,7 @@ impl HttpSlateSender {
 
 		// trivial tests for now, but will be expanded later
 		if foreign_api_version < 2 {
-			let report = format!("Other wallet reports unrecognized API format.");
+			let report = "Other wallet reports unrecognized API format.".to_string();
```
Comment (yeastplume) [inline impls/src/adapters/http.rs]: Some of this is 6 of one, half a dozen of the other kind of stuff. to_string is probably marginally more efficient.

File: impls/src/lifecycle/seed.rs [hunk 6]
```diff
@@ -203,7 +202,7 @@ impl WalletSeed {
 				 Run \"grin-wallet init\" to initialize a new wallet.",
 				seed_file_path
 			);
-			Err(ErrorKind::WalletSeedDoesntExist)?
+			Err(ErrorKind::WalletSeedDoesntExist.into())
```
Comment (yeastplume) [inline impls/src/lifecycle/seed.rs]: Why does this thing hate the ? operator so much?

**Comment by yeastplume** [comment]
There's a lot to go over here on one go, most of it is fair enough, some of it looks like someone imposing their very opinionated personal beliefs on how some things should be done, but there's likely some justification behind them that I'm not interested in bikeshedding about.

There are a couple of places where it looks like error information is being lost, other than that, all fine.

File: api/src/owner_rpc_s.rs (hunk 3 above)
Comment (yeastplume) [reply api/src/owner_rpc_s.rs]: Ah, wait, no it doesn't, never mind.

File: impls/src/lifecycle/seed.rs (hunk 6 above)
Comment (quentinlesceller) [reply impls/src/lifecycle/seed.rs]: ^^ it’s just more straightforward to return directly the error.

**Comment by quentinlesceller** [comment]
Hehe, most of it makes sense though. And I let the match on boolean untouched 😛

File: api/src/owner.rs (hunk 2 above)
Comment (quentinlesceller) [reply api/src/owner.rs]: Did it manually...

**Pull request merged**